src/optimagic/optimizers/gfo_optimizers.py

```diff
@@ -0,0 +1,864 @@
+    )
+    """Converge if the absolute change in the objective function is less than this
+    value."""
+
+    convergence_ftol_rel: NonNegativeFloat | None = None
+    """Converge if the relative change in the objective function is less than this
+    value."""
+
+    caching: bool = True
+    """Whether to cache evaluated param and function values in a dictionary for
+    lookup."""
+
+    extra_start_params: list[PyTree] | None = None
+    """List of additional start points for the optimization run.
+
+    In case of population based optimizers, the initial_population can be provided
+    via `extra_start_params`
+
+    """
+
+    warm_start: pd.DataFrame | None = None
+    """Pandas dataframe that contains score and paramter information that will be
+    automatically loaded into the memory.
+
+    example:
+
+    score       x1      x2      x...
+    0.756       0.1     0.2     ...
+    0.823       0.3     0.1     ...
+    ... ...     ...     ...
+    ... ...     ...     ...
+
+    """
+
+    verbosity: Literal["progress_bar", "print_results", "print_times"] | bool = False
+    """Determines what part of the optimization information will be printed."""
+
+    seed: int | None = None
+    """Random seed for reproducibility."""
+
+
+# ==================================================================================
+# Population Based
+# ==================================================================================
+
+
+@mark.minimizer(
+    name="gfo_pso",
+    solver_type=AggregationLevel.SCALAR,
+    is_available=IS_GRADIENT_FREE_OPTIMIZERS_INSTALLED,
+    is_global=True,
+    needs_jac=False,
+    needs_hess=False,
+    needs_bounds=True,
+    supports_parallelism=False,
+    supports_bounds=True,
+    supports_infinite_bounds=False,
+    supports_linear_constraints=False,
+    supports_nonlinear_constraints=False,
+    disable_history=False,
+)
+@dataclass(frozen=True)
+class GFOParticleSwarmOptimization(Algorithm, GFOCommonOptions):
+    """Minimize a scalar function using the Particle Swarm Optimization algorithm.
+
+    This algorithm is a Python implementation of the Particle Swarm Optimization
+    algorithm through the gradient_free_optimizers package.
+
+    Particle Swarm Optimization is a global population based algorithm.
+
+    The algorithm simulates a swarm of particles which move according to their own
+    inertia across the search space.
+    Each particle adjusts its position based on its own experience (cognitive weight)
+    and the experiences of its neighbors or the swarm (social weight), using
+    velocity updates.
+    The algorithm iteratively guides the swarm toward promising regions of the
+    search space.
+
+    The velocity of a particle is calculated by the following
+    equation:
+
+    .. math::
+        v_{n+1} = \\omega \\cdot v_n + c_k \\cdot r_1 \\cdot (p_{best}-p_n)
+        + c_s \\cdot r_2 \\cdot (g_{best} - p_n)
+
+    """
+
+    population_size: PositiveInt = 10
```

Comment:
It should be `PositiveInt | None` and the default should be None, so the `get_population_size` function can set a suitable value for the number of parameters. Same for all the other algorithms that take a population size.